core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -150,6 +164,51 @@ object DumpLogSegments {
     }
   }
 
+  private def dumpTimeIndex(file: File,
+                            indexSanityOnly: Boolean,
+                            verifyOnly: Boolean,
+                            misMatchesForIndexFilesMap: mutable.HashMap[String, List[(Long, Long)]],
+                            maxMessageSize: Int) {
+    val startOffset = file.getName().split("\\.")(0).toLong
+    val logFile = new File(file.getAbsoluteFile.getParent, file.getName.split("\\.")(0) + Log.LogFileSuffix)
+    val messageSet = new FileMessageSet(logFile, false)
+    val indexFile = new File(file.getAbsoluteFile.getParent, file.getName.split("\\.")(0) + Log.IndexFileSuffix)
+    val index = new OffsetIndex(indexFile, baseOffset = startOffset)
+    val timeIndex = new TimeIndex(file, baseOffset = startOffset)
+
+    //Check that index passes sanityCheck, this is the check that determines if indexes will be rebuilt on startup or not.
+    if (indexSanityOnly) {
+      timeIndex.sanityCheck
+      println(s"$file passed sanity check.")
+      return
+    }
+
+    var maxTimestamp = 0L
+    for(i <- 0 until timeIndex.entries) {
+      val entry = timeIndex.entry(i)
+      val position = index.lookup(entry.offset).position
+      val partialFileMessageSet: FileMessageSet = messageSet.read(position, maxMessageSize)
+      val deepIter = getIterator(partialFileMessageSet.head, isDeepIteration = true)
+      for (messageAndOffset <- deepIter)
+        maxTimestamp = math.max(maxTimestamp, messageAndOffset.message.timestamp)
+
+      if(maxTimestamp != entry.timestamp) {
+        var misMatchesSeq = misMatchesForIndexFilesMap.getOrElse(file.getAbsolutePath, List[(Long, Long)]())
+        misMatchesSeq ::=(entry.timestamp, maxTimestamp)
+        misMatchesForIndexFilesMap.put(file.getAbsolutePath, misMatchesSeq)
+      }
+      // since it is a sparse file, in the event of a crash there may be many zero entries, stop if we see one
+      if(entry.offset == 0 && i > 0)
+        return
+      if (!verifyOnly)
+        println("timestamp%s: %d offset%s: %d".format(
+          { if (maxTimestamp == entry.timestamp) "" else s"(log timestamp: $maxTimestamp)" },
```

Comment:
I was trying to list the offsets of the mismatch timestamp index entry. But the format seems wrong, I am thinking of something like the following:

When timestamp does not match.
```timestamp: 1000 (log timestamp: 2000) offset: 100```

When timestamp and offset both do not match
```timestamp: 1000 (log timestamp: 2000) offset: 100 (log offset: 200)```